### PR TITLE
[Actions] Add cache action to cache source-based dependency builds and add release versioning to source-based dependencies 

### PR DESCRIPTION
**Pull request by prashanthr05**
- Achieve faster workflow CI builds through,
  - pre-compiled vcpkg archives
  - caching source-based dependency builds

- Add release based versioning for source-based dependencies

Inspired and duplicated from
- https://github.com/robotology/idyntree/pull/655
- https://github.com/dic-iit/bipedal-locomotion-controllers/pull/32
- https://github.com/dic-iit/element_jet-cat-turbines/pull/223

### CODE REVIEW

**Comment by prashanthr05** [comment]
Is there a way to manually re-run a workflow? I can't find any.

**Comment by prashanthr05** [comment]
> Is there a way to manually re-run a workflow? I can't find any.

I would like to test the Cache feature. Is there a clean way to do this? @traversaro

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -92,72 +117,76 @@ jobs:
...
+            cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps \
+                  -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
             cmake --build . --config ${{matrix.build_type}} --target install
             
             # iDynTree
             cd ${GITHUB_WORKSPACE}
             git clone https://github.com/robotology/iDynTree
             cd iDynTree
+            git checkout ${iDynTree_TAG}
             git checkout devel
```
Comment (HosameldinMohamed) [inline .github/workflows/ci.yml]: Probably you forgot to remove this line.

**Comment by traversaro** [comment]
> 
> 
> > Is there a way to manually re-run a workflow? I can't find any.
> 
> I would like to test the Cache feature. Is there a clean way to do this instead of adding a dummy commit? @traversaro

Not that I am aware of. For testing, I personally  used by fork of the repo,  and then when everything was working I tried in the main repo I was targeting. However, I think that in this case as most of the  code is similar to the one working in iDynTree, we can also think of merging directly.

**Comment by prashanthr05** [comment]
> Not that I am aware of 

A way was shown in https://github.com/robotology/whole-body-estimators/pull/54#discussion_r397845803 ;D

**Comment by prashanthr05** [comment]
Caching was tested.
Build took
 - ubuntu: 1m 42s
 - windows: 4m 56s
 - macOS: 3m 21s

File: .github/workflows/ci.yml (hunk 1 above)
Comment (prashanthr05) [reply .github/workflows/ci.yml]: I removed this line in the recent commits https://github.com/robotology/whole-body-estimators/commit/0404ba1760e720f3d40a0c74d56f8bce0ec38f5f

**Comment by traversaro** [comment]
> But I fear the cache is using iDynTree devel. Technically, there shouldn't be a problem.

No problem, the cache produced in a PR is ignored in the main branch, so as soon as you merge it a new (correct) cache will be generated, see https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache .

**Pull request merged**